### ISSUE TITLE
CocList does not respect to `list.maxPreviewHeight`

### ISSUE THREAD

**Issue by x1314aq** [neoclide/coc.nvim#1009]
## Result from CocInfo
```
## versions

vim version: NVIM v0.3.8
node version: v12.6.0
coc.nvim version: 0.0.73-56825890dc
term: alacritty
platform: darwin

## Messages
```

## Describe the bug

The height of CocList preview window does not respect to the value of `list.maxPreviewHeight` in `coc-settings.json`

## Reproduce the bug
the content of my `init.vim`
```
" coc.nvim config
inoremap <expr> <Tab> pumvisible() ? "\<C-n>" : "\<Tab>"
inoremap <expr> <S-Tab> pumvisible() ? "\<C-p>" : "\<S-Tab>"

" close the preview window when completion is done
autocmd! CompleteDone * if pumvisible() == 0 | pclose | endif

" show signature after jump to next placeholder
autocmd User CocJumpPlaceholder call CocActionAsync('showSignatureHelp')

" Use `[c` and `]c` to navigate diagnostics
nmap <silent> [c <Plug>(coc-diagnostic-prev)
nmap <silent> ]c <Plug>(coc-diagnostic-next)

" Remap keys for gotos
nmap <silent> gd <Plug>(coc-definition)
nmap <silent> gr <Plug>(coc-references)
nmap <silent> go :<C-u>CocListResume<CR>

" enable CocList
let g:coc_enable_locationlist = 1

" Remap for rename current word
nmap <leader>rn <Plug>(coc-rename)
```
- using `<Plug>(coc-reference)` for any C functions to fill the CocList Location

## Screenshots (optional)

<img width="1280" alt="屏幕快照 2019-07-21 15 35 27" src="https://user-images.githubusercontent.com/18412635/61588390-3e194680-abcd-11e9-941c-45d8a5f542f2.png">

**Comment by x1314aq** [comment]
unfortunately, there are still incorrect cases even using the **latest** commit.

here `list.maxPreviewHeight=5`

<img width="1280" alt="屏幕快照 2019-07-22 22 33 16" src="https://user-images.githubusercontent.com/18412635/61641000-e401a980-acd0-11e9-945e-e131894f67b3.png">

**Comment by chemzqm** [comment]
@x1314aq you're using released tag, not latest commit ` * 56825890 - Release v0.0.73 (11 days ago)`
